krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -0,0 +1,73 @@
+/*
+ Copyright 2023 The Nephio Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+    "errors"
+
+    "github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+    nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+    "sigs.k8s.io/yaml"
+)
+
+type Interface interface {
+    // ParseKubeObject returns a fn sdk KubeObject; if something failed an error
+    // is returned
+    ParseKubeObject() (*fn.KubeObject, error)
+    // GetInterface returns the interface as a go struct
+    GetInterface() *nephioreqv1alpha1.Interface
+}
+
+// New creates a new parser interface
+// It expects a raw byte slice as input representing the serialized yaml file
+func New(b string) (Interface, error) {
+    i := &nephioreqv1alpha1.Interface{}
+    if err := yaml.Unmarshal([]byte(b), i); err != nil {
+        return nil, err
+    }
+    return &itfce{
+        itfce: i,
+    }, nil
+}
+
+type itfce struct {
+    itfce *nephioreqv1alpha1.Interface
+}
+
+// Marshal serializes the value provided into a YAML document based on "sigs.k8s.io/yaml".
+// The structure of the generated document will reflect the structure of the value itself.
+func (r *itfce) marshal() ([]byte, error) {
+    if r.itfce == nil {
```

Comment:
question: Will `r.itfce` be ever nil unless someone intentionally set it to nil? We may not need this method `marshal` and could just use the `yaml.Marshal(r.itfce)` if it `r.itfce` is not nil.

Reply:
the strategy changed aligned with what John was proposing. Now I added a protection as the user could set it to nil.